app/api/v1/mediator/election.py

```diff
@@ -26,12 +45,171 @@ def get_election_constants() -> Any:
+            )
+
+            return ElectionQueryResponse(
+                status=ResponseStatus.SUCCESS,
+                elections=[election],
+            )
+    except Exception as error:
+        print(sys.exc_info())
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="get election failed",
+        ) from error
+
+
+@router.put("", tags=[ELECTION])
+def submit_election(
+    election_id: Optional[str], request: SubmitElectionRequest = Body(...)
+) -> SubmitElectionResponse:
+    """
+    Submit an election.
+
+    Method expects a manifest to already be submitted or to optionally be provided
+    as part of the request body.  If a manifest is provided as part of the body
+    then it will override any cached value, however the hash must match the hash
+    contained in the CiphertextelectionContext
+    """
+    if not election_id:
+        election_id = request.election_id
+
+    if not election_id:
+        election_id = str(uuid4())
+
+    context = CiphertextElectionContext.from_json_object(request.context)
+
+    if request.manifest:
+        manifest = Manifest.from_json_object(request.manifest)
+    else:
+        manifest_query = get_manifest(context.description_hash)
+        manifest = Manifest.from_json_object(manifest_query.manifests[0])
+
+    # validate that the context was built against the correct manifest
+    if context.description_hash != manifest.crypto_hash():
+        raise HTTPException(
+            status_code=status.HTTP_412_PRECONDITION_FAILED,
+            detail="manifest hash does not match provided context hash",
+        )
+
+    election = Election(
+        election_id=election_id,
+        state=ElectionState.CREATED,
+        context=context.to_json_object(),
+        manifest=manifest.to_json_object(),
+    )
+
+    try:
+        with get_repository(CLIENT_ID, DataCollection.ELECTION) as repository:
+            _ = repository.set(write_json_object(election.dict()))
+            return SubmitElectionResponse(
+                status=ResponseStatus.SUCCESS, election_id=election_id
+            )
+    except Exception as error:
+        print(sys.exc_info())
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="Submit election failed",
+        ) from error
+
+
+@router.get("/find", tags=[ELECTION])
+def find_elections(
+    skip: int = 0, limit: int = 100, request: ElectionQueryRequest = Body(...)
+) -> ElectionQueryResponse:
+    """
+    Find elections.
+
+    Search the repository for elections that match the filter criteria specified in the request body.
+    If no filter criteria is specified the API will iterate all available data.
+    """
+    try:
+
+        filter = write_json_object(request.filter) if request.filter else {}
+        with get_repository(CLIENT_ID, DataCollection.ELECTION) as repository:
+            cursor = repository.find(filter, skip, limit)
```

Comment:
what's cool about this is it supports javascript dot notation to traverse object graphs in the key field.  e.g. `{"manifest.type": "primary"}`